src/main/java/br/com/annuum/capsicum/api/mapper/AdressMapper.java

```diff
@@ -0,0 +1,24 @@
+package br.com.annuum.capsicum.api.mapper;
+
+import br.com.annuum.capsicum.api.controller.request.AddressRequest;
+import br.com.annuum.capsicum.api.domain.Address;
+import br.com.annuum.capsicum.api.service.FindCityByIdService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AdressMapper {
+
+    @Autowired
+    private FindCityByIdService findCityByIdService;
```

Comment:
Mapper que injeta service é um caminho sem volta pro inferno. Acho que precisamos pensar bem o que é "mapear" e o que é "construir" um objeto.

Reply:
Fiquei com bastante duvida na melhor abordagem pra essas situações... Qual a melhor forma pra estruturar isso?

Reply:
Mappers geralmente servem pra transformar um ou mais objetos em outro (ou outros). Tu deveria ter todos os insumos pra fazer o mapeamento e não ir atrás deles dentro do Mapper. Se for pensar bem: qual é a principal função desse enpoint? Buscar uma cidade pelo seu id, eu diria. E isso está sendo feito dentro de um mapper, ou seja, a princiapal lógica está no mapper. 

Nesse caso, eu teria feito um mapper que recebe um request e transforma em um Address sem cidade e, na service, setaria a cidade.

```
City city = service.getCity(request.getId)
Address address = mapper.map(request)
   .setCity(city);
```